neo/utils.py

```diff
@@ -0,0 +1,856 @@
+
+    if 'name' not in kwargs:
+        kwargs['name'] = 'epoch'
+    if 'labels' not in kwargs:
+        # this needs to be changed to '%s_%i' % (kwargs['name'], i) for i in range(len(times))]
+        # when labels become unicode
+        kwargs['labels'] = [
+            ('%s_%i' % (kwargs['name'], i)).encode('ascii') for i in range(len(times))]
+
+    ep = neo.Epoch(times=times, durations=durations, **kwargs)
+
+    ep.annotate(**event1.annotations)
+
+    if attach_result:
+        segment.epochs.append(ep)
+        segment.create_relationship()
+
+    return ep
+
+
+def match_events(event1, event2):
+    """
+    Finds pairs of Event entries in event1 and event2 with the minimum delay,
+    such that the entry of event1 directly precedes the entry of event2.
+    Returns filtered two events of identical length, which contain matched
+    entries.
+
+    Parameters:
+    -----------
+    event1, event2: neo.Event
+        The two Event objects to match up.
+
+    Returns:
+    --------
+    event1, event2: neo.Event
+        Event objects with identical number of events, containing only those
+        events that could be matched against each other. A warning is issued if
+        not all events in event1 or event2 could be matched.
+    """
+    # load the full event if a proxy object has been given as an argument
+    if isinstance(event1, neo.io.proxyobjects.EventProxy):
+        event1 = event1.load()
+    if isinstance(event2, neo.io.proxyobjects.EventProxy):
+        event2 = event2.load()
+
+    id1, id2 = 0, 0
+    match_ev1, match_ev2 = [], []
+    while id1 < len(event1) and id2 < len(event2):
+        time1 = event1.times[id1]
+        time2 = event2.times[id2]
+
+        # wrong order of events
+        if time1 >= time2:
+            id2 += 1
+
+        # shorter epoch possible by later event1 entry
+        elif id1 + 1 < len(event1) and event1.times[id1 + 1] < time2:
+            # there is no event in 2 until the next event in 1
+            id1 += 1
+
+        # found a match
+        else:
+            match_ev1.append(id1)
+            match_ev2.append(id2)
+            id1 += 1
+            id2 += 1
+
+    if id1 < len(event1):
+        warnings.warn(
+            'Could not match all events to generate epochs. Missed '
+            '%s event entries in event1 list' % (len(event1) - id1))
+    if id2 < len(event2):
+        warnings.warn(
+            'Could not match all events to generate epochs. Missed '
+            '%s event entries in event2 list' % (len(event2) - id2))
+
+    event1_matched = _event_epoch_slice_by_valid_ids(
+        obj=event1, valid_ids=match_ev1)
+    event2_matched = _event_epoch_slice_by_valid_ids(
+        obj=event2, valid_ids=match_ev2)
+
+    return event1_matched, event2_matched
+
+
+def cut_block_by_epochs(block, properties=None, reset_time=False):
+    """
+    This function cuts Neo Segments in a Neo Block according to multiple Neo
```

Comment:
There is no need to prefix the object names by "Neo" in this context!

Reply:
I went through the whole file and removed all unnecessary prefixes.